applications/FluidDynamicsApplication/custom_elements/stokes_3D_twofluid.h

```diff
@@ -371,6 +371,38 @@ class Stokes3DTwoFluid
                 Output = 0.5*inner_prod(data.stress, strain);
             }
         }
+        else if(rVariable == EQ_STRAIN_RATE) //compute effective strain rate
+        {
+                Output = mp_constitutive_law->GetValue(rVariable, Output);
+                // std::cout <<"StrainRate: "<<Output <<std::endl;
+                return;          
+        } 
+        else if(rVariable == EFFECTIVE_VISCOSITY) //compute the heat flux per unit volume induced by the shearing
+        {
+            const unsigned int NumNodes = 4;
+            const unsigned int Dim = 3;
+            const unsigned int strain_size = 6;
+            
+            
+             double distance_center = 0.0;
+            for(unsigned int i=0; i<GetGeometry().size(); i++)
+                distance_center += GetGeometry()[i].FastGetSolutionStepValue(DISTANCE);
+            distance_center/=static_cast<double>(GetGeometry().size());
+            
+            const double air_nu = GetProperties()[DYNAMIC_VISCOSITY]; //ATTENTION: not using here the real visosity of air
```

Comment:
this could go inside the following if

Reply:
Also, I would try to use a `const` properties, for example
```suggestion
            const Properties& r_properties = GetProperties();
            const double air_nu = r_properties[DYNAMIC_VISCOSITY]; //ATTENTION: not using here the real visosity of air
```
Since this would avoid a race condition in the (admittedly very rare) case that one reaches this point without having initialized the viscosity value in properties.